webapp/networks/utils.ts

```diff
@@ -1,14 +1,18 @@
+import { isValidUrl } from 'utils/url'
 import { type Chain } from 'viem'
 import { defineChain } from 'viem/utils'
 
-export const overrideRpcUrl = function (chain: Chain, rpcUrl?: string) {
-  const isValidCustomSepoliaRpc = !!rpcUrl && rpcUrl.startsWith('https')
-  if (isValidCustomSepoliaRpc) {
+export const updateRpcUrls = function (chain: Chain, rpcUrlEnv?: string) {
+  if (typeof rpcUrlEnv !== 'string') {
+    return chain
+  }
```

Comment:
Why is this needed?

Reply:
in case a malformed configuraion. Then, it returns the chains as-is (without changes).

The `chain` is a read-only object (It can't be updated), so in order to update it we need to create a new one, and use that instead. By returning the OG without modifications, we can just import everywhere from this file (instead of the definition in `viem`) and it will work.